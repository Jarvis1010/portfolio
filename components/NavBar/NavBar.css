.NavBar{
    box-sizing: border-box;
    overflow: hidden;
    position:fixed;
    font-family:'Helvetica Neue',Helvetica,Arial,sans-serif;
    line-height: 1.42857143;
    left:0;
    right:0;
    padding:1em;
    display:flex;
    font-size:14px;
    background-color:#eaeaea;
    z-index: 1000;
    flex:1;
    color:#777;
    & [class*="NavItems-"]>*:hover{
        color:#333 !important;
    }

    @media screen and (max-width: 600px) {
        justify-content: space-between;
    }

    & a{
        text-decoration: none;
        color:inherit;
        &:hover{
            text-decoration: none;
        }
    }
}
.inverse{
    background-color:#222;
    color:#9d9d9d;
    & [class*="NavItems-"]>*:hover{
            color:white !important;
        
    }
}
.img{
    width:auto;
    max-height: 22px;
}
.disableClick{
    pointer-events: none;
}
.top{
    top:0;
}
.bottom{
    bottom:0;
}