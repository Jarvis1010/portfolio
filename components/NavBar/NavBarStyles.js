import React from 'react';
import styled from 'styled-components';

export const NavWrapper = styled.nav`
    box-sizing: border-box;
    overflow: hidden;
    position:fixed;
    font-family:'Helvetica Neue',Helvetica,Arial,sans-serif;
    line-height: 1.42857143;
    left:0;
    right:0;
    padding:1em;
    display:flex;
    font-size:14px;
    background-color:${props => (props.inverse ? '#222' : '#eaeaea')};
    z-index: 1000;
    flex:1;
    color:${props => (props.inverse ? '#9d9d9d' : '#777')};
  
    & a{
        text-decoration: none;
        color:inherit;
        &:hover{
            text-decoration: none;
        }
    }
    @media (max-width: 600px) {
        justify-content: space-between;
    }

    ${props => (props.bottom ? 'bottom:0;' : 'top:0;')}
`;

export const Icon = styled.img`
    width:auto;
    max-height:22px;
`;

export const IconAnchor = styled.a`
    ${props => props.disabled && 'pointer-events:none;'}
`;
