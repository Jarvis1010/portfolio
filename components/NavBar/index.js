import React, { Fragment } from "react";
import styled from "styled-components";
import Link from "next/link";

const NavWrapper = styled.nav`
  background: ${props => props.theme.primaryColor || "grey"};
  min-height: 50px;
  position: relative;
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-end;
  color: ${props => props.theme.white || "white"};
  position: sticky;
  z-index: 6;
  top: 0;
`;

const NavItem = styled.span`
  display: flex;
  color: ${props => props.theme.white || "white"};
  white-space: nowrap;
  & a {
    text-decoration: none;
    color: ${props => props.theme.white || "white"};
  }
  align-self: center;
  margin: ${props => props.theme.spacing8 || "1rem"};
`;

const BrandName = styled.h2`
  font-family: ${props => props.theme.fontFamily || "sans"};
  font-size: ${props => props.theme.fontSize18 || "18px"};
  min-width: auto;
  margin-left: ${props => props.theme.spacing8 || "1rem"};
  white-space: nowrap;
`;

const BrandLink = styled.a`
  align-items: center;
  border: 0;
  color: ${props => props.theme.white || "white"};
  display: flex;
  cursor: pointer;
  text-decoration: none;
  flex-grow: 10;
  margin-right: auto;

  &:hover {
    color: ${props => props.theme.white || "white"};
    text-decoration: none;
  }
`;

const Logo = styled.img`
  max-width: ${props => props.size || "30px"};
  height: auto;
  margin: ${props => props.theme.spacing8 || "1rem"};
`;

const Brand = ({ brandName = "", src = "", href }) => (
  <Link href={href}>
    <BrandLink>
      {src != "" && <Logo src={src} alt={`${brandName} logo`} />}
      {brandName != "" && <BrandName>{brandName}</BrandName>}
    </BrandLink>
  </Link>
);

const NavSlider = styled.div`
  min-height: 50px;
  display: flex;
  flex-flow: row wrap;
  font-size: ${props => props.theme.fontSize18 || "18px"};

  @media (max-width: ${props => props.theme.breakWidth480 || "480px"}) {
    display: ${props => (props.hidden ? "none" : "flex")};
    font-size: ${props => props.theme.fontSize13 || "13px"};
    width: 100%;
    order: 50;
    background: ${props => props.theme.secondaryColor || "grey"};
  }
`;

const Hamburger = styled.button`
  display: none;
  text-align: center;
  text-decoration: none;
  font-size: ${props => props.theme.fontSize18 || "18px"};
  color: ${props => props.theme.white || "white"};
  border: 1px solid ${props => props.theme.white || "white"};
  cursor: pointer;
  background: ${props => props.theme.primaryColor || "inherit"};

  @media screen and (max-width: ${props =>
      props.theme.breakWidth480 || "480px"}) {
    display: inline;
    margin: auto ${props => props.theme.spacing8 || "1rem"};
  }
`;

class Toggle extends React.Component {
  state = { hidden: true };
  updateState = val => this.setState({ hidden: val });
  render() {
    return this.props.children([this.state.hidden, this.updateState]);
  }
}

const NavCollapse = ({ children }) => {
  return (
    <Toggle>
      {([hidden, updateHidden]) => (
        <Fragment>
          <Hamburger onClick={() => updateHidden(!hidden)}>â˜°</Hamburger>
          <NavSlider hidden={hidden}>{children}</NavSlider>
        </Fragment>
      )}
    </Toggle>
  );
};

export { NavWrapper, NavItem, NavSlider, NavCollapse, Brand };
