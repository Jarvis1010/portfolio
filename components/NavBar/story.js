import React from "react";
import { storiesOf } from "@storybook/react";
import { NavWrapper, NavItem, Brand, NavSlider, NavCollapse } from "./index";
import Link from "next/link";

storiesOf("NavBar", module)
  .add("NavWrapper", () => <NavWrapper />)
  .add("WithBrandOnly", () => (
    <NavWrapper>
      <Brand brandName={"Blue Moon Backup"} href={"/"} />
    </NavWrapper>
  ))
  .add("withIconOnly", () => (
    <NavWrapper>
      <Brand
        src={
          "https://bluemoonbackup.com/wp-content/themes/Karma/images/_global/custom-logo-9.png"
        }
        href="/"
      />
    </NavWrapper>
  ))
  .add("withIconandBrand", () => (
    <NavWrapper>
      <Brand
        brandName={"Blue Moon Backup"}
        src={
          "https://bluemoonbackup.com/wp-content/themes/Karma/images/_global/custom-logo-9.png"
        }
        href="/"
      />
    </NavWrapper>
  ))
  .add("withSlider", () => (
    <NavWrapper>
      <Brand
        brandName={"Blue Moon Backup"}
        src={
          "https://bluemoonbackup.com/wp-content/themes/Karma/images/_global/custom-logo-9.png"
        }
        href="/"
      />
      <NavSlider>
        <NavItem>
          <Link href="/">
            <a>test</a>
          </Link>
        </NavItem>
        <NavItem>
          <Link href="/">
            <a>test</a>
          </Link>
        </NavItem>
        <NavItem>
          <Link href="/">
            <a>test</a>
          </Link>
        </NavItem>
      </NavSlider>
      <NavItem>Testing</NavItem>
    </NavWrapper>
  ))
  .add("withCollapse", () => (
    <NavWrapper>
      <Brand
        brandName={"Blue Moon Backup"}
        src={
          "https://bluemoonbackup.com/wp-content/themes/Karma/images/_global/custom-logo-9.png"
        }
        href="/"
      />
      <NavCollapse>
        <NavItem>
          <Link href="/">
            <a>test</a>
          </Link>
        </NavItem>
        <NavItem>
          <Link href="/">
            <a>test</a>
          </Link>
        </NavItem>
        <NavItem>
          <Link href="/">
            <a>test</a>
          </Link>
        </NavItem>
      </NavCollapse>
    </NavWrapper>
  ));
