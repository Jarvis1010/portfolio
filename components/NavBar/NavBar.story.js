import { storiesOf, action } from '@kadira/storybook';
import NavBar from './NavBar';
import React from 'react';
import NavCollapse from '../NavCollapse/NavCollapse';
import NavItems from '../NavItems/NavItems';
import NavForm from '../NavForm/NavForm';

const iconImage = 'https://ip.ldschurch.org/app/images/integration-point.svg';

storiesOf('NavBar', module)
  .add('default', () => (
    <NavBar inverse={true} img={iconImage}>
      <NavCollapse>
        <NavItems>
          <a href="#">Home</a>
          <a>Library</a>
          <NavForm onClick={action('Search Button Clicked')} />
        </NavItems>
        <NavItems>
          <div>Welcome Anon!</div>
          <div>logoff</div>
        </NavItems>
      </NavCollapse>
    </NavBar>
  ))
  .add('fixed bottom and inverse', () => (
    <NavBar img={iconImage} inverse={true} fixed="bottom">
      <NavCollapse>
        <NavItems>
          <div>Home</div>
          <div>Library</div>
        </NavItems>
        <NavItems>
          <div>Welcome Anon!</div>
          <div>logoff</div>
        </NavItems>
      </NavCollapse>
    </NavBar>
  ))
  .add('No collapse', () => (
    <NavBar img={iconImage}>
      <NavItems forceRow={true}>
        <div>Home</div>
        <div>Library</div>
      </NavItems>
    </NavBar>
  ))
  .add('No image, collapse, and left only', () => (
    <NavBar>
      <div>Brand</div>
      <NavCollapse>
        <NavItems>
          <div>Home</div>
          <div>Library</div>
        </NavItems>
      </NavCollapse>
    </NavBar>
  ))
  .add('Collapse and right only', () => (
    <NavBar img={iconImage}>
      <NavCollapse>
        <div />
        <NavItems>
          <div>Welcome Anon!</div>
          <div>logoff</div>
        </NavItems>
      </NavCollapse>
    </NavBar>
  ));
