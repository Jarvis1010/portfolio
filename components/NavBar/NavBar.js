import React from 'react';
import { NavWrapper, Icon, IconAnchor } from './NavBarStyles';

export const NavBar = ({ className, children, img, href = '', ...props }) => (
  <NavWrapper {...props}>
    {img &&
      <IconAnchor disabled={href === ''} href={href}>
        <Icon src={img} />
      </IconAnchor>}
    {children}
  </NavWrapper>
);

export default NavBar;
