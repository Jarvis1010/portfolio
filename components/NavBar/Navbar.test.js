import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import { NavWrapper, Brand, NavSlider, NavItem } from "../NavBar";

const theme = {
  white: "white",
  fontFamily: "Palatino",
  primaryColor: "dodgerblue"
};

describe("Navbar Components Render", () => {
  describe("Navwrapper Renders", () => {
    it("Renders NavWrapper", () => {
      const wrapper = shallow(<NavWrapper />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it("Renders NavWrapper with theme", () => {
      const wrapper = shallow(<NavWrapper theme={theme} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe("NavSlider Renders", () => {
    it("Renders NavSlider", () => {
      const wrapper = shallow(<NavSlider />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it("Renders NavSlider with theme", () => {
      const wrapper = shallow(<NavSlider theme={theme} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe("NavItem Renders", () => {
    it("Renders NavSlider", () => {
      const wrapper = shallow(<NavItem />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it("Renders NavSlider with theme", () => {
      const wrapper = shallow(<NavItem theme={theme} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe("Brand Renders", () => {
    it("Renders Brand", () => {
      const allProps = { brandName: "BrandName", src: "/", href: "#" };
      const missingBrandName = { src: "/", href: "#" };
      const missingSrc = { brandName: "BrandName", href: "#" };
      [allProps, missingBrandName, missingSrc].forEach(props => {
        const wrapper = shallow(<Brand {...props} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
    });
  });
});
