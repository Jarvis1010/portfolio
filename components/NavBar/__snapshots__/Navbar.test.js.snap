// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Components Render Brand Renders Renders Brand 1`] = `
<Link
  href="#"
>
  <styled.a>
    <styled.img
      alt="BrandName logo"
      src="/"
    />
    <styled.h2>
      BrandName
    </styled.h2>
  </styled.a>
</Link>
`;

exports[`Navbar Components Render Brand Renders Renders Brand 2`] = `
<Link
  href="#"
>
  <styled.a>
    <styled.img
      alt=" logo"
      src="/"
    />
  </styled.a>
</Link>
`;

exports[`Navbar Components Render Brand Renders Renders Brand 3`] = `
<Link
  href="#"
>
  <styled.a>
    <styled.h2>
      BrandName
    </styled.h2>
  </styled.a>
</Link>
`;

exports[`Navbar Components Render NavItem Renders Renders NavSlider 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  color: ",
          [Function],
          ";
  white-space: nowrap;
  & a {
    text-decoration: none;
    color: ",
          [Function],
          ";
  }
  align-self: center;
  margin: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Navbar Components Render NavItem Renders Renders NavSlider with theme 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  color: ",
          [Function],
          ";
  white-space: nowrap;
  & a {
    text-decoration: none;
    color: ",
          [Function],
          ";
  }
  align-self: center;
  margin: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "fontFamily": "Palatino",
      "primaryColor": "dodgerblue",
      "white": "white",
    }
  }
/>
`;

exports[`Navbar Components Render NavSlider Renders Renders NavSlider 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": false,
        "rules": Array [
          "
  min-height: 50px;
  display: flex;
  flex-flow: row wrap;
  font-size: ",
          [Function],
          ";

  @media (max-width: ",
          [Function],
          ") {
    display: ",
          [Function],
          ";
    font-size: ",
          [Function],
          ";
    width: 100%;
    order: 50;
    background: ",
          [Function],
          ";
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Navbar Components Render NavSlider Renders Renders NavSlider with theme 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": false,
        "rules": Array [
          "
  min-height: 50px;
  display: flex;
  flex-flow: row wrap;
  font-size: ",
          [Function],
          ";

  @media (max-width: ",
          [Function],
          ") {
    display: ",
          [Function],
          ";
    font-size: ",
          [Function],
          ";
    width: 100%;
    order: 50;
    background: ",
          [Function],
          ";
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "fontFamily": "Palatino",
      "primaryColor": "dodgerblue",
      "white": "white",
    }
  }
/>
`;

exports[`Navbar Components Render Navwrapper Renders Renders NavWrapper 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  background: ",
          [Function],
          ";
  min-height: 50px;
  position: relative;
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-end;
  color: ",
          [Function],
          ";
  position: sticky;
  z-index: 6;
  top: 0;
",
        ],
      },
      "displayName": "styled.nav",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "nav",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Navbar Components Render Navwrapper Renders Renders NavWrapper with theme 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  background: ",
          [Function],
          ";
  min-height: 50px;
  position: relative;
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-end;
  color: ",
          [Function],
          ";
  position: sticky;
  z-index: 6;
  top: 0;
",
        ],
      },
      "displayName": "styled.nav",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "nav",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "fontFamily": "Palatino",
      "primaryColor": "dodgerblue",
      "white": "white",
    }
  }
/>
`;
